### PR TITLE
setting up the people history model

### PR DESCRIPTION
**Pull request by kevinverhoff**
# Pull Request

## Summary & Motivation
Setting up an INT view in order to avoid separate tableau views that are doing similar things. Adding items to the int view that cover most of the downstream reporting needs

[//]: # "When merged, this pull request will..."

## Self-review

- [x] Update **due date**, **assignee**, and **priority** on our
      [TEAMster Asana Project](https://app.asana.com/0/1205971774138578/1205971926225838)

  _If this is a same-day request, please flag that in Slack_

- [x] <kbd>Format</kbd> has been run on all modified files

- [ ] Ensure you are using the `union_dataset_join_clause()` macro for queries that employ any
      models from these datasets:
  - deanslist
  - edplan
  - iready
  - pearson
  - powerschool
  - renlearn
  - titan

## Troubleshooting

- [SqlFluff Rules Reference](https://docs.sqlfluff.com/en/stable/rules.html)
- [Trunk](https://teamschools.github.io/teamster/CONTRIBUTING/#trunk)
- [dbt](https://teamschools.github.io/teamster/CONTRIBUTING/#dbt-cloud_1)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207049037257016

### CODE REVIEW

File: src/dbt/kipptaf/models/people/intermediate/int_people__historic_data_as_of_april30_annually.sql [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            extract(year from date_day) as academic_year,
+            if(
+                extract(year from date_day) = {{ var("current_academic_year") }}
+                and extract(month from current_date('{{ var("local_timezone") }}'))
+                not in (5, 6),
+                current_date('{{ var("local_timezone") }}'),
+                date((extract(year from date_day) + 1), 4, 30)
+            ) as effective_date,
+        from {{ ref("utils__date_spine") }}
+        where extract(month from date_day) = 7 and extract(day from date_day) = 1
```
Comment (cbini) [inline src/dbt/kipptaf/models/people/intermediate/int_people__historic_data_as_of_april30_annually.sql:15]: I'm phasing out `utils__date_spine`. Use `generate_date_array` instead:
```
select 
  effective_date,
  extract(year from effective_date) - 1 as academic_year,
from unnest(generate_date_array("2003-04-30", date(2024, 4, 30), INTERVAL 1 YEAR)) as effective_date
```

**Review by cbini — CHANGES_REQUESTED** [review]
- [ ] fix dbt build errors
- [ ] can the tableau views be merged?

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__finance_accounting_people_model.sql [hunk 2]
```diff
@@ -1,19 +1,4 @@
...
-            }} as academic_year,
-        from {{ ref("utils__date_spine") }}
-        where
-            extract(month from date_day) = 4
-            and extract(day from date_day) = 30
-            and extract(year from date_day) >= 2016
-    ),
-
     additional_earnings_clean as (
         select distinct
```
[the remark this answers is not in the document]
Comment (kevinverhoff) [reply src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__finance_accounting_people_model.sql:3]: The distinct is necessary - there are some duplicated rows on the source file

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__finance_accounting_people_model.sql [hunk 3]
```diff
@@ -32,82 +17,58 @@ with
...
     ye.years_at_kipp_total as years_at_kipp_total_current,
+
+    ae.additional_earnings_total as additional_earnings_summed,
+
+    null as original_salary_upon_hire,
+    null as is_currently_certified_nj_only,
+
     ye.years_teaching_at_kipp
     + coalesce(cw.years_teaching_in_njfl, 0)
     + coalesce(cw.years_teaching_outside_njfl, 0) as years_teaching_total_current,
```
[the remark this answers is not in the document]
Comment (kevinverhoff) [reply src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__finance_accounting_people_model.sql:61]: good call - I'll update that

File: src/dbt/kipptaf/models/people/intermediate/int_people__historic_data_as_of_april30_annually.sql
[the remark this answers is not in the document]
Comment (kevinverhoff) [reply src/dbt/kipptaf/models/people/intermediate/int_people__historic_data_as_of_april30_annually.sql:1]: lol okay

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__finance_accounting_people_model.sql [hunk 4]
```diff
@@ -32,82 +17,58 @@ with
...
+    hd.management_position_indicator as is_manager,
+    hd.worker_original_hire_date as original_hire_date,
+    hd.worker_rehire_date as rehire_date,
+    hd.worker_termination_date as termination_date,
+    hd.overall_score as most_recent_pm_score,
+    hd.overall_tier as most_recent_pm_tier,
 
     ly.business_unit_home_name as last_year_business_unit,
     ly.job_title as last_year_job_title,
+    ly.historic_salary as last_year_salary,
```
[the remark this answers is not in the document]
Comment (kevinverhoff) [reply src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__finance_accounting_people_model.sql:50]: check

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__finance_accounting_people_model.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (cbini) [reply src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__finance_accounting_people_model.sql:3]: interesting, do you have an example record? might be worth adding to the staging model if it's not something fixable

**Comment by kevinverhoff** [comment]
I think we're ready to go now!

**Review by cbini — APPROVED** [review]

**Pull request merged**